File: Module/ALittleIDE/Other/GameLibrary/CEngine/JSScript/CEngine/Version/VersionSystemWindows.js
{
if (typeof ALittle === "undefined") ALittle = {};


if (ALittle.VersionSystem === undefined) throw new Error(" extends class:ALittle.VersionSystem is undefined");
ALittle.VersionSystemWindows = JavaScript.Class(ALittle.VersionSystem, {
	Ctor : function(account_name, module_name) {
		this._install_name = "Install.exe";
	},
	RefreshVersion : function() {
	},
	InstallImpl : function(install_name) {
		ALittle.System_InstallProgram(install_name + " /silent /norestart");
		ALittle.System_ForceExit();
	},
	Install : function(install_name) {
		if (install_name === undefined) {
			install_name = ALittle.File_BaseFilePath() + this._update_path + this._install_name;
		}
		let updater = ALittle.NewObject(ALittle.LoopFunction, ALittle.VersionSystemWindows.InstallImpl.bind(undefined, install_name), 1, 0, 1);
		updater.Start();
	},
}, "ALittle.VersionSystemWindows");

}

File: ALittle/Module/CEngine/Script/CEngine/Version/VersionSystemWindows.lua
-- ALittle Generate Lua
module("ALittle", package.seeall)

local ___thispath = select('1', ...):match("(.+[/\\]).+$") or ""
local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittle.VersionSystem, " extends class:ALittle.VersionSystem is nil")
VersionSystemWindows = Lua.Class(ALittle.VersionSystem, "ALittle.VersionSystemWindows")

function VersionSystemWindows:Ctor(account_name, module_name)
	___rawset(self, "_install_name", "Install.exe")
end

function VersionSystemWindows.RefreshVersion()
end

function VersionSystemWindows.InstallImpl(install_name)
	System_InstallProgram(install_name .. " /silent /norestart")
	System_ForceExit()
end

function VersionSystemWindows:Install(install_name)
	if install_name == nil then
		install_name = File_BaseFilePath() .. self._update_path .. self._install_name
	end
	local updater = LoopFunction(Lua.Bind(VersionSystemWindows.InstallImpl, install_name), 1, 0, 1)
	updater:Start()
end

